### PR TITLE
[Task 1872345] migrate the code to create storage from track1 to track2

### PR DESCRIPTION
**Pull request by andxu**
[AB#1872345](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1872345)

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/CreateArmStorageAccountForm.java [hunk 1]
```diff
@@ -328,68 +307,29 @@ protected void okPressed() {
...
+                .region(region)
+                .kind(kind)
+                .resourceGroup(resourceGroup)
+                .performance((Performance) performanceCombo.getData(performanceCombo.getText()))
+                .redundancy((Redundancy) replicationComboBox.getData(replicationComboBox.getText()))
+                .subscription(subscription)
+                .accessTier(Optional.ofNullable(accessTierComboBox).map(t -> (AccessTier) accessTierComboBox.getData(accessTierComboBox.getText())).orElse(null))
+                .build();
+        new Thread(this.onCreate).start();
         super.okPressed();
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/CreateArmStorageAccountForm.java:333]: remove new thrad

Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/CreateArmStorageAccountForm.java:333]: Resolve it in new PR

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/actions/CreateArmStorageAccountAction.java [hunk 2]
```diff
@@ -26,8 +40,37 @@ public CreateArmStorageAccountAction(StorageModule storageModule) {
...
+                        "Creating storage account " + createStorageAccountForm.getStorageAccount().getName() + "...", new Runnable() {
+                            @Override
+                            public void run() {
+                                EventUtil.executeWithLog(STORAGE, CREATE_STORAGE_ACCOUNT, (operation) -> {
+                                    createStorageAccount(createStorageAccountForm.getStorageAccount());
+                                            storageModule.load(false);
+                                        }, (e) ->
+                                                AzureTaskManager.getInstance().runLater(() ->
+                                                        PluginUtil.displayErrorDialog(PluginUtil.getParentShell(), Messages.err,
+                                                                "An error occurred while creating the storage account: " + e.getMessage())
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/actions/CreateArmStorageAccountAction.java:54]: use AzureMessager.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/actions/CreateArmStorageAccountAction.java [hunk 3]
```diff
@@ -26,8 +40,37 @@ public CreateArmStorageAccountAction(StorageModule storageModule) {
     public void actionPerformed(NodeActionEvent e) {
         SignInCommandHandler.requireSignedIn(PluginUtil.getParentShell(), () -> {
             CreateArmStorageAccountForm createStorageAccountForm = new CreateArmStorageAccountForm(PluginUtil.getParentShell(), null, null);
-            createStorageAccountForm.setOnCreate(() -> storageModule.load(false));
+            createStorageAccountForm.setOnCreate(() -> {
+                AzureTaskManager.getInstance().runInBackground(
+                        "Creating storage account " + createStorageAccountForm.getStorageAccount().getName() + "...", new Runnable() {
+                            @Override
+                            public void run() {
+                                EventUtil.executeWithLog(STORAGE, CREATE_STORAGE_ACCOUNT, (operation) -> {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/actions/CreateArmStorageAccountAction.java:48]: is it possible to use AzureOperation annotation?

Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/actions/CreateArmStorageAccountAction.java:48]: shall not related to track2. move to future PR

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/actions/CreateArmStorageAccountAction.java (hunk 2 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/actions/CreateArmStorageAccountAction.java:54]: shall not related to track2. move to future PR

**Review by Flanker32 — APPROVED** [review]
Please resolve the comments in later PR

**Pull request merged**